src/OpenTelemetry.ResourceDetectors.Container/ContainerResourceDetector.cs

```diff
@@ -132,45 +143,102 @@ private static string RemovePrefixAndSuffixIfNeeded(string input, int startIndex
         return input.Substring(startIndex, endIndex - startIndex);
     }
 
+    private string? ExtractK8sContainerId()
+    {
+        try
+        {
+            var baseUrl = this.k8sMetadataFetcher.GetServiceBaseUrl();
```

Comment:
We try to extract K8s result first - which is correct - but it utilize HTTP request which can be relatively slow. Can we make extract code fail early (with no warning, as it is perfectly fine - we don't want any warning on non-K8s environments) if any required environment variable is not set or any file is not available?

Reply:
Done, please tell me what you think.